### PR TITLE
fix: use native currency price

### LINKED ISSUE #1283: Fetch native currency USD price from CoinGecko
**Description**
For native currency prices we are fetching the asset wrapped version so we can use the same endpoint. But this returns the wrong price, as there are different prices for wrapped and native assets.

#### Definition of done
- [ ] Description is created (Task creator)
- [ ] Code is complete (Developer)
- [ ] Scope of test and what was done section is defined in task (Developer)
- [ ] Review passed (Code reviewer)
- [ ] Verified manually (QA)
- [ ] Test cases created/updated if necessary (QA)
- [ ] Change successfully merged
- [ ] Release notes added if needed

**Proposal**
Fetch native currencies USD price from a different endpoint, so the currency prices matches CoinGecko UI.

### PR DESCRIPTION
**Pull request by berteotti**
# Summary

Fixes #1283

Use Coingecko to fetch native currency prices

# To Test

1. Simulate trade with native currency
- [ ] Check if native currency USD price is equal to Coingecko price

### CODE REVIEW

File: src/hooks/useUSDValue.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export function useUSDPrice(tokenAmount?: TokenAmount) {
   }, [chainId, tokenAmount, stablecoin, tradeExactInUniswapV2])
 }
 
-export function useCoingeckoUSDPrice(token?: Token) {
+export function useCoingeckoUSDPrice(token?: Token, isNativeCurrency: boolean = false) {
```
Comment (karczuRF) [inline src/hooks/useUSDValue.ts:90]:
```suggestion
export function useCoingeckoUSDPrice(token?: Token, isNativeCurrency = false) {
```

File: src/hooks/useUSDValue.ts [hunk 2]
```diff
@@ -194,8 +198,8 @@ export function useUSDValue(tokenAmount?: TokenAmount) {
   return useGetPriceQuote({ price: price, tokenAmount: tokenAmount })
 }
 
-export function useCoingeckoUSDValue(tokenAmount?: TokenAmount) {
-  const coingeckoUsdPrice = useCoingeckoUSDPrice(tokenAmount?.token)
+export function useCoingeckoUSDValue(tokenAmount?: TokenAmount, isNativeCurrency: boolean = false) {
```
Comment (karczuRF) [inline src/hooks/useUSDValue.ts:201]:
```suggestion
export function useCoingeckoUSDValue(tokenAmount?: TokenAmount, isNativeCurrency = false) {
```

File: src/utils/coingecko.ts [hunk 3]
```diff
@@ -80,6 +91,25 @@ export async function getUSDPriceQuote(params: CoinGeckoUsdPriceParams): Promise
   return response.json()
 }
 
+export async function getUSDPriceCurrencyQuote(
+  params: CoinGeckoUsdPriceCurrencyParams
+): Promise<CoinGeckoUsdQuote | null> {
+  const { chainId } = params
+
+  const nativeCurrency = COINGECKO_NATIVE_CURRENCY[chainId]
+  if (nativeCurrency == null) {
```
Comment (karczuRF) [inline src/utils/coingecko.ts:100]:
```suggestion
  if (!nativeCurrency) {
```

File: src/utils/coingecko.ts [hunk 4]
```diff
@@ -80,6 +91,25 @@ export async function getUSDPriceQuote(params: CoinGeckoUsdPriceParams): Promise
...
+    // Unsupported
```
Comment (karczuRF) [inline src/utils/coingecko.ts:101]: You mean unsupported network? What about more detailed comment?

**Review by Mi-Lan — APPROVED** [review]
looks good!

**Comment by niemam29** [comment]
Tested and looks good to me!

**Review by adamazad — APPROVED** [review]
Besides the comments from @karczuRF, this should be merged

**Comment by berteotti** [comment]
@karczuRF your comments are fixed in [4261b00](https://github.com/levelkdev/swapr-dapp/pull/1345/commits/4261b001d178ee1157fa8d0f45d8cb2b10d5eae5). Can you review it again please?

**Review by karczuRF — APPROVED** [review]

**Comment by niemam29** [comment]
Retested and looks good to me!

**Pull request merged**